pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -18,7 +18,16 @@ in rec {
   #
   # * https://nixos.org/nix/manual/#ssec-derivation
   #   Explanation about derivations in general
-  mkDerivation =
+  mkDerivation = fnOrAttrs:
+    let
+      fn =
+        if builtins.isFunction fnOrAttrs
+        then fnOrAttrs
+        else _: fnOrAttrs;
+      this = mkDerivation_ (transform: mkDerivation (transform fn)) (fn { inherit this; });
+    in this;
+
+  mkDerivation_ = remakeDerivation:
```

Comment:
I would make this take `overrideAttrs` not `remakeDerivation`, so we can do all the "extensions and fixed points" combinatorics above.

Reply:
I've made it look like `makeExtensible` instead and documented the differences. That also means I'm adding `overrideAttrs` in `makeDerivationExtensible`, just after applying the user's function.
So this is closer to the `lib` functions and moves all of the overriding stuff outside `mkDerivation_`.